File: 257. 二叉树的所有路径/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def traversal(self, node, s, ans):
        if not node:
            return
        if not node.left and not node.right:
            s += str(node.val)
            ans.append(s)
            return
        s += str(node.val) + "->"
        self.traversal(node.left, s, ans)
        self.traversal(node.right, s, ans)
        return

    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        ans = []
        if not root:
            return ans
        s = ""
        self.traversal(root, s, ans)
        return ans

File: 257. 二叉树的所有路径/Solution.cpp
#include<bits/stdc++.h>

using namespace std;

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    // 如果这里的s引用的是地址那么需要进行回溯，如果不是引用则不需要pop回溯
    void traversal(TreeNode* node, string s, vector<string>& ans){ 
        if(node == NULL) return;
        if(node->left == NULL && node->right == NULL){
            s += to_string(node->val);
            ans.push_back(s);
            return;
        }
        s += to_string(node->val) + "->";
        traversal(node->left, s, ans);
        traversal(node->right, s, ans);
        return;
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        string s = "";
        vector<string> ans;
        traversal(root, s, ans);
        return ans;
    }
};